### PR TITLE
added McqRenderers tests

### PR DESCRIPTION
**Pull request by tussharrane**

### CODE REVIEW

File: src/Components/McqRenderer/McqRenderer.test.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+import McqRenderer from './McqRenderer';
+import store from '../../store/store';
```
Comment (kkyusufk) [inline src/Components/McqRenderer/McqRenderer.test.js:5]: lets make use of this package
https://www.npmjs.com/package/redux-mock-store

**Review by kkyusufk — CHANGES_REQUESTED** [review]

File: src/Components/McqRenderer/McqRenderer.test.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+		render(
+			<Provider store={store}>
+				<McqRenderer data={data} showContent={true} />
+			</Provider>
+		);
+		store.dispatch({ type: SET_PAUSED, payload: true });
+		expect(screen.queryByTestId(/dimmer/i)).toBeInTheDocument();
+	});
+
+	test('when lessonPaused is false, Dimmer is removed', () => {
```
Comment (bios1337) [inline src/Components/McqRenderer/McqRenderer.test.js:120]: These two tests can be combined to reduce the LOC:
[true, false].forEach(lessonPaused => {
   test(`when lessonPaused is ${lessonPaused}, Dimmer is ${lessonPaused? 'shown' : 'not shown'}`, () => {
		render(
			<Provider store={store}>
				<McqRenderer data={data} showContent={true} />
			</Provider>
		);
		store.dispatch({ type: SET_PAUSED, payload: lessonPaused });
		expect(screen. queryAllByTestId(/dimmer/i).length).toBe(lessonPause? 1 : 0);
	});
})

File: src/Components/McqRenderer/McqRenderer.test.js [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+		expect(screen.queryByAltText(/incorrect/i)).not.toBeInTheDocument();
+		expect(onEndFunction).toBeCalledTimes(1);
+	});
+
+	[true, false].forEach((lessonPaused) => {
+		test(`when lessonPaused is ${lessonPaused}, Dimmer is ${
+			lessonPaused ? 'shown' : 'not Shown'
+		}`, () => {
+			const modifiedStore = mockStore({
```
Comment (bios1337) [inline src/Components/McqRenderer/McqRenderer.test.js:118]: nit-pick: its probably a good idea to make a variable at the top of the test, some tests have many test cases in that case you won't have to initialize mockStore again and again. You can just use 
`initialState.lessonPaused = true`

Note: You have two stores (one mocked and other the original) maybe we can avoid these two sources of truths

Comment (tussharrane) [reply src/Components/McqRenderer/McqRenderer.test.js:118]: So, I can update the state by just updating the initialState variable ?

Comment (bios1337) [reply src/Components/McqRenderer/McqRenderer.test.js]: Yes, I think so

Comment (tussharrane) [reply src/Components/McqRenderer/McqRenderer.test.js]: Yes, it works.

**Review by bios1337 — APPROVED** [review]
👍🏽

**Review by kkyusufk — APPROVED** [review]

**Pull request merged**